platform/engineering/request-for-comment/2024-09-23-evidence-collection-and-polling.md

```diff
@@ -0,0 +1,136 @@
+# RFC: Benefits Evidence and Ancillary Documents Handling and Technical Details
+
+- Date: 2024-09-23
+- Related Issue: 
+
+## Background
+Evidence and ancillary forms filled out on veterans/users behalf, are collected, generated, stored, and sent in many different ways across products/forms.
+
+For a better user experience, and for the sake of less technical complexity, some amount of technical and architectural decisions should be agreed upon and implemented across all products and forms. Ideally, where possible, sharing code to reduce complexity and drift. But at the very least should operate similarly. This RFC aims to describe and gain concensus on the full evidence and ancillary form lifecycle. 
+
+## Definitions
+Evidence: Any document uploaded by a user, that vets-api is responisble for transmitting down- or up- stream. Examples could include a medical document, buddy statement, scanned-in filled-out form, digitally filled-out form, and other various documents. 
+
+Ancillary Document: Any document that has an official VA designated form number, that we fill-out and transmit somewhere, based on information the end-user enters into the frontend. Depending on the form/situation, the end-user may or may not know they filled this form out, or that one was filled out on their behalf. Examples include form 21-4142, form 21-0781, and other forms that we take information and stamp onto a PDF to transmit.
+
+Documents: May be used to refer to both evidence and ancillary documents collectivly.
+
+
+## Motivation
+
+* Reduced code complexity
+* Align on shared strategy
+* Consistent fronend experience for the end users
+* Elimination of silent failure (through status polling)
+
+
+## Design
+
+### Frontend Accepting of Benefits documents
+#### Validation
+Validation should be kept to a minimum on the frontend only to the most obvious issues, such as:
+* Filesize not exceeding X
+* Document extension supported
+* Document not password protected (or password collected if so, since we NEED to check this here)
+
+All other checks should be done by the system downstream that is ultimately responsible for accepting the file or not. We should not try to maintain separate lists of requirements that can get out of sync. We should, after initial validation above, check the file against a validation endpoint for the downstream system. 
```

Comment:
We could try to word this as the "furthest synchronous downstream source of truth" with the thought being that a synchronous document validation response is needed to alert the user.
This would mean that the accepting service would be required to do as much validation as possible when there is a lack of usable downstream validation.
I agree that there is ideally a single source of truth on valid documents, we should do what we can to alert the veteran that their document will not be accepted.